docs/adr/0004-withdraw-expiry.rst

```diff
@@ -0,0 +1,91 @@
+  once the withdraw request expires, node (A) will remove the expired total withdraw state and locked withdraw will be unlocked.
+
+Examples of Option 2
+====================
+
+Simple:
+-------
+
+- Node (A)'s balance = 100 tokens
+- Node (A) requests a withdraw of 50 tokens. In this case, usable balance of node (A) is 50 (assuming it did not send/receive transfers to/from node (B)).
+  `total_withdraw=50`.
+- Node (B) does not confirm with withdraw.
+- Node (A) sends node (B) a `WithdrawExpired` message. Node (A) only had 1 pending withdraw state of 50.
+  Before sending `WithdrawExpired`, node (A) clears the expired withdraw state and `total_withdraw` is set to 0 because that is the only withdraw
+  state that was requested.
+
+
+Complex:
+--------
+
+- Node (A)'s balance = 100 tokens
+- Node (A) requests a withdraw of 10 tokens. In this case, usable balance of node (A) is 50 (assuming it did not send/receive transfers to/from node (B)).
+  `total_withdraw=10`.
+- Node (A) requests a withdraw of another 10 tokens. `total_withdraw=20`.
+- Node (A) requests a withdraw of another 20 tokens. `total_withdraw=40`.
+- Node (A) withdraw states look as follows
+```
+[
+  {
+    "total_withdraw": 10,
+  },
+  {
+    "total_withdraw": 20,
+  }
+  {
+    "total_withdraw": 30,
+  }
+]
+```
+- Node (B) does not confirm any withdraw.
+- Node (A) sends node (B) a `WithdrawExpired` message for the first withdraw request which had a `total_withdraw=10`.
+  Before sending `WithdrawExpired`, node (A) clears the expired withdraw state `total_withdraw=10`. However, that doesn't affect node (A)'s `total_withdraw`
+  because the latest one is 30 and stays 30.
+- Node (A) sends node (B) a `WithdrawExpired` again for the second withdraw request which had a `total_withdraw=20`. Same as before applies.
+- Node (A) sends node (B) a `WithdrawExpired` again for the third and last withdraw request which had a `total_withdraw=30`. Similar to the simple case,
+  this was the only withdraw state that is pending, therefore expiring this withdraw state means that the `total_withdraw` of node (A) goes back to zero.
+
+
+Why Nonce was included in a non-balance proof message
+=====================================================
+
+Previously, only balance-proof messages had a nonce. The nonce value provides certain guarantees one of which is the fact that we are
+able to process messages in the exact order in which they were sent.
+Though `WithdrawRequest`, `WithdrawConfirmation` and `WithdrawExpired` messages are not balance-proof messages,
+a nonce value was required to make sure that these messages are also processed in the order in which they were sent. To accomodate for this change,
+the nonce field has to be decoupled from being based on the latest balance proof to a value that is maintained for the two sides of the channel state
+so that the next value is requested by either balance-proof based messages or withdraw messages.
```

Comment:
Does the nonce need to be the same? Can't BP nonces be different than WithdrawExpiry nonces?

Reply:
We wanted to have the same value to maintain the order of processing all messages.